lib/setUserInfo.js

```diff
@@ -0,0 +1,64 @@
+import Sequelize from 'sequelize'
+import moment from 'moment'
+import getToken from './getToken'
+import models from '../models'
+import {ADMIN_USERGROUP_ID} from '../config'
+
+export default async function setUserInfo(req) {
+	if (req.get('Authorization')) {
+		let userToken = getToken(req)
+
+		let userData
+		try {
+			userData = await models.User.findOne({
+				where: {
+					authToken: userToken
+				},
+				include: [
+					{
+						model: models.UserGroupMembership,
+						as: 'user',
+						include: [
+							{
+								model: models.UserGroup,
+								as: 'userGroup'
+							}
+						]
+					}
+				]
+			})
+		} catch (err) {
+			console.log('Error getting current user from database: ' + err)
+		}
+
+		if (!userData) {
+			return req
+		}
+
+		// simplify list of user's groups
+		userData.isAdmin = false
+		let userGroups = []
+		userData.user.forEach((ug) => {
+			userGroups.push({
+				id: ug.dataValues.userGroup.dataValues.id,
+				name: ug.dataValues.userGroup.dataValues.name
+			})
+			// while we're here, may as well check if the user is an admin
+			if (ug.dataValues.userGroup.dataValues.id === ADMIN_USERGROUP_ID) {
+				userData.isAdmin = true
+			}
+		})
+		userData.userGroups = userGroups
+
+		if (userData.authToken && userData.authToken == userToken && moment(userData.tokenExpiration) > moment()) {
+			// this is a valid user with a non-expired token, so set some data on their request and return it
+			req.user = userData // attach the user to the req
```

Comment:
we should now be able to access all the user's data directly on the `req` object. I'd suggest that we make the properties that we care about available on the top level of `req.user`, as I did here with userGroups.